### PR TITLE
log output to a callback function

### PR DESCRIPTION
**Pull request by xiaozhou**
Enable the mobile client to receive log output. See `log/test/logtest.go` for example usage.

### CODE REVIEW

File: log/log.go [hunk 1]
```diff
@@ -329,6 +337,12 @@ func SetFileName(name string) {
 	std.name = name
 }
 
+func SetOutput(out Output) {
+	std.rw.Lock()
+	defer std.rw.Unlock()
+	std.out = out
```
Comment (rdcel) [inline log/log.go:343]: The places that use "std.out" (access it) to print don't seem to be grabbing the reader-lock of std.rw. 
Actually nobody is doing std.rw.Rlock().  Probably tiny corner case as most users either don't change
these values or only do them at initialization in the single-threaded part of their code.  Just wondering 
about this lock's usage.

File: log/test/logtest.go [hunk 2]
```diff
@@ -1,19 +1,34 @@
...
 	"flag"
+	"fmt"
 
 	"github.com/celer-network/goutils/log"
 )
 
+type TestOutput struct {
+}
+
+func (cb TestOutput) Write(output []byte) (n int, err error) {
```
Comment (rdcel) [inline log/test/logtest.go:20]: nit: it's a nicer Go pattern to use a pointer for the object receiver instead of a struct
in case somebody copy/pastes this code

File: log/log.go (hunk 1 above)
Comment (xiaozhou) [reply log/log.go:343]: right, I should use std.mu.Lock(), will make another pr to fix it
